### PR TITLE
[wallet/desktop] fix: cosign form is not presented for multisig tx iss#1863

### PR DESCRIPTION
**Pull request by OlegMakarenko**
## Problem

- In some specific cases where the account is a cosigner of multiple multisig-accounts wallets are not present button to sign a transaction.
- Some accounts are missing in multisig tree.

## Solution

- Fetch full multisig tree.
- Added new multisig view.

image

### CODE REVIEW

File: src/views/modals/ModalTransactionCosignature/ModalTransactionCosignatureTs.ts [hunk 1]
```diff
@@ -317,7 +319,8 @@ export class ModalTransactionCosignatureTs extends Vue {
                 if (c) {
                     return (
                         c.plain() === this.currentAccount.address ||
-                        (this.multisigAccountGraph && MultisigService.isAddressInMultisigTree(this.multisigAccountGraph, c.plain()))
+                        (this.multisigAccountGraph && MultisigService.isAddressInMultisigTree(this.multisigAccountGraph, c.plain())) ||
+                        (this.rootMultisigAccountGraph && MultisigService.isAddressInMultisigTree(this.rootMultisigAccountGraph, c.plain()))
                     );
```
Comment (Jaguar0625) [inline src/views/modals/ModalTransactionCosignature/ModalTransactionCosignatureTs.ts:324]: consider test for new branch

Comment (OlegMakarenko) [reply src/views/modals/ModalTransactionCosignature/ModalTransactionCosignatureTs.ts:324]: Thanks. I found another issue, working on fix now. I will reopen PR later.

File: src/store/Account.ts [hunk 2]
```diff
@@ -584,24 +578,40 @@ export default {
...
-                const currentMultisigAccountsInfo = MultisigService.getMultisigInfoFromMultisigGraphInfo(currentMultisigAccountGraphInfo);
-                const rootAddressIndex = currentMultisigAccountsInfo.length - 1;
-                const rootAddress = currentMultisigAccountsInfo[rootAddressIndex].accountAddress;
+                const rootMultisigAccountGraph = new Map(currentMultisigAccountGraphInfo.multisigEntries);
+
+                for (const [currentLevel, multisigAccountInfos] of currentMultisigAccountGraphInfo.multisigEntries) {
+                    if (currentLevel < 0) {
+                        for (const multisigAccountInfo of multisigAccountInfos) {
+                            const fetchedMultisigAccountGraphInfo = await repositoryFactory
+                                .createMultisigRepository()
```
Comment (Jaguar0625) [inline src/store/Account.ts:587]: use forEach ?

File: __tests__/components/AccountMultisigGraph.spec.ts [hunk 3]
```diff
@@ -165,26 +142,80 @@ describe('components/AccountMultisigGraph', () => {
...
-        expect(component.multisigGraphTree[0].info.cosignatoryAddresses[1].address).toBe(cosig2Address);
-        expect(component.multisigGraphTree[0].children[0].info.accountAddress.address).toBe(cosig1Address);
-        expect(component.multisigGraphTree[0].children[1].info.accountAddress.address).toBe(cosig2Address);
-        expect(component.multisigGraphTree[0].children[0].children[0].info.accountAddress.address).toBe(cosig11Address);
-        expect(component.multisigGraphTree[0].children[1].children[0].info.accountAddress.address).toBe(cosig21Address);
-        expect(component.multisigGraphTree[0].children[0].selected).toBe(true);
+        expect(mockUpdateGraphConfig).toBeCalledTimes(1);
+        expect(mockCenterGraph).toBeCalledTimes(1);
+        expect(component.isGraphModalShown).toBe(true);
+        expect(graphViewElement.exists()).toBeTruthy();
```
Comment (Jaguar0625) [inline __tests__/components/AccountMultisigGraph.spec.ts:219]: why truthy vs checking explicit value ?

File: src/store/Account.ts (hunk 2 above)
Comment (OlegMakarenko) [reply src/store/Account.ts:587]: Do you mean map it into array of `Promise`s?

Comment (Jaguar0625) [reply src/store/Account.ts:587]: nevermind, looking again, i think what you have is fine 👍

File: __tests__/components/AccountMultisigGraph.spec.ts (hunk 3 above)
Comment (OlegMakarenko) [reply __tests__/components/AccountMultisigGraph.spec.ts]: Since `.exists()` returns `bool`, yes it makes sense to check explicit value. Thanks, updated.

**Review by Jaguar0625 — APPROVED** [review]

**Pull request merged**